### PR TITLE
Create test_generalized_fish_game.py writing tests using pytest for generalized_fish_game.py file

### PR DESCRIPTION
**Pull request by Jaydon2005**
Tests plotting functions to ensure they produce plots. Test cases cover a variety of scenarios and edge cases to ensure robust testing of the generalized_fish_game,py

### CODE REVIEW

File: msdbook/tests/test_generalized_fish_game.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+from msdbook.generalized_fish_game import (
+    inequality,
+    plot_uncertainty_relationship,
+    plot_solutions,
+    fish_game,
+    hrvSTR
+)
+
```
Comment (erexer) [inline msdbook/tests/test_generalized_fish_game.py:11]: should have 2 lines between imports and functions as well as 2 lines between functions according to [PEP 8](https://peps.python.org/pep-0008/#blank-lines)

File: msdbook/tests/test_generalized_fish_game.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    nCnstr = 1
+
+    objs, cnstr = fish_game(vars, additional_inputs, N, tSteps, nObjs, nCnstr)
+
+    assert len(objs) == nObjs, "Incorrect number of objective values"
+    assert len(cnstr) == nCnstr, "Incorrect number of constraint values"
+    assert np.all(np.isfinite(objs)), "Objective values should be finite"
+    assert np.all(np.isfinite(cnstr)), "Constraint values should be finite"
+
+@pytest.mark.mpl_image_compare
```
[the remark this answers is not in the document]
Comment (Jaydon2005) [reply msdbook/tests/test_generalized_fish_game.py:51]: The code above requires pytest-mpl because of the @pytest.mark.mpl_image_compare decorator. It is used to compare images in tests that include matplotlib plots.

[the remark this answers is not in the document]
Comment (erexer) [reply msdbook/tests/test_generalized_fish_game.py:58]: Can you make sure the `pyproject.toml` has the right requirements? You should also merge in the `dev` branch since this branch is out of date.

[the remark this answers is not in the document]
Comment (Jaydon2005) [reply msdbook/tests/test_generalized_fish_game.py:58]: I updated the pyproject.toml so it has the right requirements.

File: pyproject.toml [hunk 3]
```diff
@@ -38,6 +38,8 @@ dependencies = [
     "scipy>=1.13.1",
     "seaborn>=0.13.2",
     "statsmodels>=0.14.2",
+    "mpl-image-compare",
```
Comment (erexer) [inline pyproject.toml:41]: This should be in the `optional-dependencies` `dev` section.

File: pyproject.toml [hunk 4]
```diff
@@ -38,6 +38,8 @@ dependencies = [
...
+    # other dependencies
```
Comment (erexer) [inline pyproject.toml:42]: you should remove this since it's unnecessary

**Comment by erexer** [comment]
closes https://github.com/IMMM-SFA/msd_uncertainty_ebook/issues/122

File: msdbook/tests/test_generalized_fish_game.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+)
+
+
+def test_inequality():
+    b = 0.5
+    m = 0.9
+    h = 0.1
+    K = 1000
+    result = inequality(b, m, h, K)
+    expected = (b**m) / (h * K) ** (1 - m)
```
Comment (erexer) [inline msdbook/tests/test_generalized_fish_game.py:19]: Since this is duplicating exactly what the function does, it's not an effective test. You should hardcode this value instead.

File: msdbook/tests/test_generalized_fish_game.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    Inputs = [0.5]
+    vars = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6]
+    input_ranges = [[0, 1]]
+    output_ranges = [[0, 1]]
+
+    
+    result = hrvSTR(Inputs, vars, input_ranges, output_ranges)
+    print("HRVSTR output:", result)  # Check the actual output
+    # Adjust the expected based on correct calculations
+    expected = [result[0]]  # Change this to the expected value if necessary
```
Comment (erexer) [inline msdbook/tests/test_generalized_fish_game.py:33]: This should be a hardcoded value. Since we're trying to determine if the function is producing incorrect outputs, and we know what the output should be given certain inputs, we should check for that output.

File: msdbook/tests/test_generalized_fish_game.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        "Previous_Prey",
+        "0.1", "0.2", "0.3", "0.4", "0.5", "0.6", "0.7", "0.8", "0.9"
+    ]
+    N = 10
+    tSteps = 100
+    nObjs = 5
+    nCnstr = 1
+
+    
+    objs, cnstr = fish_game(vars, additional_inputs, N, tSteps, nObjs, nCnstr)
```
Comment (erexer) [inline msdbook/tests/test_generalized_fish_game.py:49]: This should also check for a hardcoded value. Since we're trying to determine if the function is producing incorrect outputs, and we know what the output should be given certain inputs, we should check for that output.

File: msdbook/tests/test_generalized_fish_game.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+    
+    assert len(objs) == nObjs, "Incorrect number of objective values"
+    assert len(cnstr) == nCnstr, "Incorrect number of constraint values"
+    assert np.all(np.isfinite(objs)), "Objective values should be finite"
+    assert np.all(np.isfinite(cnstr)), "Constraint values should be finite"
+
+
+@pytest.mark.mpl_image_compare
```
Comment (erexer) [inline msdbook/tests/test_generalized_fish_game.py:58]: Can you update the docs to give instructions on how to run these tests.

File: pyproject.toml (hunk 3 above)
Comment (erexer) [reply pyproject.toml]: ah I was going off the [`pyproject.toml`](https://github.com/IMMM-SFA/msd_uncertainty_ebook/blob/dev/pyproject.toml#L67) in dev which uses a separate `test` `optional-dependencies` section and it looks like you're using `dev`. So when I installed `pip install -e ".[test]"` it was failing because pytest wasn't installed.
